src/slate-toolbar/modal-dialog.tsx

```diff
@@ -34,29 +54,155 @@ export const ModalDialog: React.FC<IProps> = (props) => {
-    if ((e.key === "Escape") || (e.keyCode === 27)) return handleCancelClick();
+    if (!isMenuOpen) {
+      if ((e.key === "Enter") || (e.keyCode === 13)) return handleOkClick();
+      if ((e.key === "Escape") || (e.keyCode === 27)) return handleCancelClick();
+    }
   };
-  const handleValueChange = (index: number) => (e: any) => {
-    const newArr = [...inputValues];
-    newArr[index] = e.target.value;
-    setInputValue(newArr);
+  const handleValueChange = (name: string, value: string) => {
+    setFieldValues(_values => {
+      const newValues = cloneDeep(fieldValues);
+      newValues[name] = value;
+      return newValues;
+    });
   };
 
+  function renderInput(field: IField, index: number, inline: boolean) {
+    const inputType = field.type === "checkbox" ? field.type : "text";
+    const inValue = field.type === "checkbox"
+                      ? undefined
+                      : fieldValues[field.name] || "";
+    const inChecked = field.type === "checkbox"
+                        ? fieldValues[field.name] === "true"
+                        : undefined;
+    const eventValue = (e: React.ChangeEvent<HTMLInputElement>) => {
+      return field.type === "checkbox" ? String(e.target.checked) : e.target.value;
+    };
+    return (
+      <input
+        id={field.name}
+        name={field.name}
+        className={inline ? "" : "full-row"}
+        onChange={e => handleValueChange(e.target.name, eventValue(e))}
+        type={inputType}
+        value={inValue}
+        checked={inChecked}
+        size={field.charSize}
+        ref={index===0 ? input1Ref : undefined}
+      />
+    );
+  }
+
+  function renderCheckbox(field: IField, index: number, inline: boolean) {
+    return (
+      // cf. https://dev.to/proticm/styling-html-checkboxes-is-super-easy-302o
+      <label className="checkbox">
+        {renderInput(field, index, inline)}
+        {field.label && <span>{field.label}</span>}
+      </label>
+    );
+  }
+
+  function renderSelect(field: IField, index: number, inline: boolean) {
+    // cf. https://github.com/JedWatson/react-select/issues/1322#issuecomment-605614912
+    const customStyles = {
+      container: (provided: any) => ({
+        ...provided,
+        padding: '5px'
+      }),
+      control: (provided: any, state: any) => ({
+        ...provided,
+        background: '#fff',
+        borderColor: '#9e9e9e',
+        borderRadius: '2px',
+        minHeight: '30px',
+        height: '30px',
+        boxShadow: state.isFocused ? null : null,
+      }),
+      valueContainer: (provided: any) => ({
+        ...provided,
+        height: '30px',
+        padding: '0 6px'
+      }),
+
+      input: (provided: any) => ({
+        ...provided,
+        margin: '0px',
+      }),
+      indicatorSeparator: (state: any) => ({
+        display: 'none',
+      }),
+      indicatorsContainer: (provided: any) => ({
+        ...provided,
+        height: '30px',
+      }),
+    };
```

Comment:
This select could use a `cursor: pointer;` somewhere so the mouse behaves like the checkbox above it.

Reply:
Done